src/edit/index.vue

```diff
@@ -1,9 +1,116 @@
 <template>
-  <p>Custom element template</p>
+  <div class="tce-api-video">
+    <element-placeholder
+      v-if="isEmpty"
+      :is-focused="isFocused"
+      :is-disabled="isDisabled"
+      name="Api Video component"
+      icon="mdi-video-image"
+      active-placeholder="Use toolbar to upload the video"
+      active-icon="mdi-arrow-up" />
+    <div v-else>
+      <tce-overlay v-if="error || didUploadFail" type="error">
+        <v-icon color="error">mdi-alert</v-icon>
+        {{ error || 'Video upload failed. Please try again.' }}
+      </tce-overlay>
+      <tce-overlay v-if="infoMessage">
+        <v-progress-circular indeterminate color="info" class="mr-4" />
+        {{ infoMessage }}
+      </tce-overlay>
```

Comment:
Can we reuse `TceOverlay`? We could move `error` and `didUploadFail` behind the same computed property to reduce some noise here (i.e. default error message could be returned from computed).